controllers/metal3.io/baremetalhost_controller_test.go

```diff
@@ -2000,3 +2001,50 @@ func TestInvalidBMHCanBeDeleted(t *testing.T) {
 		return host.Status.Provisioning.State == metal3v1alpha1.StateDeleting && len(host.Finalizers) == 0
 	})
 }
+
+func TestCredentialsFromSecret(t *testing.T) {
+	cases := []struct {
+		name     string
+		input    corev1.Secret
+		expected bmc.Credentials
+	}{
+		{
+			name:     "empty",
+			input:    corev1.Secret{},
+			expected: bmc.Credentials{},
+		},
+		{
+			name: "clean",
+			input: corev1.Secret{
+				Data: map[string][]byte{
+					"username": []byte("username"),
+					"password": []byte("password"),
+				},
+			},
+			expected: bmc.Credentials{
+				Username: "username",
+				Password: "password",
+			},
+		},
+		{
+			name: "newline",
```

Comment:
Not sure if you want to cover all the Unicode cases, but I think it could be useful to have at least the case like

```
[]byte("   username"),
[]byte("password   "),
```

Reply:
I added a case for non-newline whitespace, but I don't think it's necessary to retest strings.TrimSpace() in our test suite.